code/game/objects/items/weapons/glowstick.dm

```diff
@@ -52,3 +52,57 @@
 #undef GLOW_GREEN
 #undef GLOW_RED
 #undef GLOW_BLUE
+
+/obj/item/clothing/accessory/glowstick //Maybe later convert the rest over
+	name = "glowstick"
+	desc = "A glowstick filled with luminescent liquid. It has a string on it so you can wear it."
+	icon_state = "glowstick"
+	_color = "glowstick" //for setting accessory states
+	color = rgb(255, 255, 255) //for coloring the accessory
+
+/obj/item/clothing/accessory/glowstick/attack_self(mob/user)
+	to_chat(user, "<span class='notice'>You crack \the [src] and it lights up.")
+	set_light(2, l_color = color)
+
+/obj/item/clothing/accessory/glowstick/suicide_act(mob/user)
+	user.visible_message("<span class='danger'>[user] is breaking open \the [src.name] and eating the liquid inside! It looks like \he's  trying to commit suicide!</span>")
+	return (TOXLOSS)
+
+
+/obj/item/clothing/accessory/glowstick/phazon
+	name = "phazon glowstick"
+	desc = "A glowstick filled with phazon material that will change colors upon agitation. It has a string on it so you can wear it."
+	var/currentcolor
+
+/obj/item/clothing/accessory/glowstick/phazon/New()
+	..()
+	colorchange()
+
+/obj/item/clothing/accessory/glowstick/phazon/pickup(mob/user)
+	user.callOnFace |= "\ref[src]"
```

Comment:
I'm 99.5% sure this line is redundant due to the line directly under it.
I assume it's copypasted, though, so not really your fault